### PR TITLE
[BEAM-1441] Fix size check for windows and improve error message

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

R: @aaltay PTAL
cc: @charlesccychen

### CODE REVIEW

File: sdks/python/apache_beam/io/filesystem.py [hunk 1]
```diff
@@ -269,8 +270,9 @@ class FileMetadata(object):
   """
   def __init__(self, path, size_in_bytes):
     assert isinstance(path, basestring) and path, "Path should be a string"
-    assert isinstance(size_in_bytes, int) and size_in_bytes >= 0, \
-        "Size of bytes should be greater than equal to zero"
+    assert isinstance(size_in_bytes, numbers.Number) and size_in_bytes >= 0, \
```
Comment (charlesccychen) [inline sdks/python/apache_beam/io/filesystem.py]: Can you use `isinstance(size_in_bytes, (int, long))` instead and add a comment that Windows uses longs here?  I think using numbers.Number may be too general (e.g. floats match).  (and can you remove the `import numbers` above?)

File: sdks/python/apache_beam/io/filesystem.py [hunk 2]
```diff
@@ -269,8 +270,9 @@ class FileMetadata(object):
...
+        "Size of bytes should be greater than zero %s with type %s"  % (
```
Comment (charlesccychen) [inline sdks/python/apache_beam/io/filesystem.py]: This comment right now incorrectly suggests that the printed `type(size_in_bytes)` is the desired type.  How about the following: `"Invalid value for size_in_bytes: %s (of type %s)"`

**Comment by charlesccychen** [comment]
Thanks! LGTM after a few nits.

File: sdks/python/apache_beam/io/filesystem.py (hunk 2 above)
Comment (sb2nov) [inline sdks/python/apache_beam/io/filesystem.py]: Done.

File: sdks/python/apache_beam/io/filesystem.py (hunk 1 above)
Comment (sb2nov) [inline sdks/python/apache_beam/io/filesystem.py]: Done.

**Comment by sb2nov** [comment]
R: @aaltay PTAL

The error was one of the HighMemoryError issues, should re-run from the new commit.

**Comment by charlesccychen** [comment]
Thanks, LGTM.

File: sdks/python/apache_beam/io/filesystem.py [hunk 3]
```diff
@@ -269,8 +269,9 @@ class FileMetadata(object):
   """
   def __init__(self, path, size_in_bytes):
     assert isinstance(path, basestring) and path, "Path should be a string"
-    assert isinstance(size_in_bytes, int) and size_in_bytes >= 0, \
-        "Size of bytes should be greater than equal to zero"
+    assert isinstance(size_in_bytes, (int, long)) and size_in_bytes >= 0, \
+        "Size of bytes should be greater than zero %s with type %s"  % (
```
Comment (charlesccychen) [inline sdks/python/apache_beam/io/filesystem.py]: Was not addressed: 	
This comment right now incorrectly suggests that the printed type(size_in_bytes) is the desired type. How about the following: `"Invalid value for size_in_bytes: %s (of type %s)"`

Comment (sb2nov) [inline sdks/python/apache_beam/io/filesystem.py]: Sorry forgot to push that.

**Comment by sb2nov** [comment]
@aaltay high memory issue again should be safe to ignore given your other PR

**Pull request closed without merging**